samza-core/src/main/java/org/apache/samza/zk/ZkProcessorLatch.java

```diff
@@ -66,8 +67,19 @@ public void await(long timeout, TimeUnit timeUnit) throws TimeoutException {
 
   @Override
   public void countDown() {
-    // create persistent (should be ephemeral? Probably not)
+    // create persistent node
     String path = zkUtils.getZkClient().createPersistentSequential(latchPath + "/", participantId);
-    LOGGER.debug("ZKProcessorLatch countDown created " + path);
+    LOG.debug("ZKProcessorLatch countDown created " + path);
+  }
+
+  @Override
+  public void close() {
+    try {
+      if (zkUtils != null)
+        zkUtils.close();
```

Comment:
We should not be closing zkUtils here since it is being instantiated outside of this class and passed as constructor argument.

Reply:
I understand this concern. I can move close() to the CoordinationUitls. The only downside for it is that we can close all the utilities only together.